plasma_framework/test/src/exits/payment/PaymentChallengeIFEInputSpent.test.js

```diff
@@ -0,0 +1,357 @@
+const { buildUtxoPos } = require('../../../helpers/positions.js');
+const { createInputTransaction, createInFlightTx, getOutputId } = require('../../../helpers/ife.js');
+const { PaymentTransactionOutput, PaymentTransaction } = require('../../../helpers/transaction.js');
+const { spentOnGas, computeNormalOutputId } = require('../../../helpers/utils.js');
+
+contract('PaymentChallengeIFEInputSpent', ([_, alice, inputOwner, outputOwner, challenger]) => {
+    const IN_FLIGHT_EXIT_BOND = 31415926535; // wei
+    const PIGGYBACK_BOND = 31415926535;
+    const MIN_EXIT_PERIOD = 60 * 60 * 24 * 7; // 1 week
+    const DUMMY_INITIAL_IMMUNE_VAULTS_NUM = 0;
+    const INITIAL_IMMUNE_EXIT_GAME_NUM = 1;
+    const YOUNGEST_POSITION_BLOCK = 1000;
+    const INFLIGHT_EXIT_YOUNGEST_INPUT_POSITION = buildUtxoPos(YOUNGEST_POSITION_BLOCK, 0, 0);
+    const ETH = constants.ZERO_ADDRESS;
+    const BLOCK_NUMBER = 5000;
+    const MAX_INPUT_SIZE = 4;
+
+    before('deploy and link with controller lib', async () => {
+        const startInFlightExit = await PaymentStartInFlightExit.new();
+        const piggybackInFlightExit = await PaymentPiggybackInFlightExit.new();
+        const challengeInFlightExitNotCanonical = await PaymentChallengeIFENotCanonical.new();
+        const challengeIFEInputSpent = await PaymentChallengeIFEInputSpent.new();
+
+        await PaymentInFlightExitRouter.link('PaymentStartInFlightExit', startInFlightExit.address);
+        await PaymentInFlightExitRouter.link('PaymentPiggybackInFlightExit', piggybackInFlightExit.address);
+        await PaymentInFlightExitRouter.link('PaymentChallengeIFENotCanonical', challengeInFlightExitNotCanonical.address);
+        await PaymentInFlightExitRouter.link('PaymentChallengeIFEInputSpent', challengeIFEInputSpent.address);
+
+        this.exitIdHelper = await ExitId.new();
+        this.stateTransitionVerifierAccept = await StateTransitionVerifierAccept.new();
+    });
+
+    describe('challenge in-flight exit input spent', () => {
+        // This is the transaction whose output is the input piggyback in the IFE.
+        function buildInputTx() {
+            const tx = createInputTransaction(
+                [buildUtxoPos(2000, 4, 3), buildUtxoPos(2000, 10, 2)],
+                inputOwner,
+                334455,
+            );
+
+            const txBytes = web3.utils.bytesToHex(tx.rlpEncoded());
+            const outputIndex = 0;
+
+            return {
+                tx,
+                txBytes,
+                outputIndex,
+                utxoPos: buildUtxoPos(3000, 5, outputIndex),
+            };
+        }
+
+        // Sets up an IFE tx using inputTx as an input, starts the IFE and piggybacks on the input.
+        const buildPiggybackInputData = async (inputTx) => {
+            const outputAmount = 997;
+
+            const firstInput = createInputTransaction([buildUtxoPos(BLOCK_NUMBER, 3, 0)], outputOwner, 334455);
+            const firstInputUtxoPos = buildUtxoPos(3000, 66, 0);
```

Comment:
Minor: lol, `firstInput` is using input with utxo that is of block 5000 (BLOCK_NUMBER) while itself is at block number 3000 XDD.